ftml/src/render/text/mod.rs

```diff
@@ -101,13 +107,29 @@ impl Render for TextRender {
         page_info: &PageInfo,
         settings: &WikitextSettings,
     ) -> String {
-        self.render_partial_direct(
-            &tree.elements,
+        self.render_partial_direct(RenderPartial {
+            elements: &tree.elements,
             page_info,
             settings,
-            &tree.table_of_contents,
-            &tree.footnotes,
-            &tree.bibliographies,
-        )
+            table_of_contents: &tree.table_of_contents,
+            footnotes: &tree.footnotes,
+            bibliographies: &tree.bibliographies,
+            wikitext_len: tree.wikitext_len,
+        })
     }
 }
+
+/// Helper structure to pass in values for `render_partial_direct()`.
+///
+/// This exists because otherwise the function would take an excessive
+/// number of parameters, which Clippy dislikes.
+#[derive(Debug)]
+struct RenderPartial<'a> {
+    elements: &'a [Element<'a>],
+    page_info: &'a PageInfo<'a>,
+    settings: &'a WikitextSettings,
+    table_of_contents: &'a [Element<'a>],
+    footnotes: &'a [Vec<Element<'a>>],
+    bibliographies: &'a BibliographyList<'a>,
+    wikitext_len: usize,
+}
```

Comment:
Adding one extra paramater to this function means clippy complains about too many, so I created a wrapper structure for the `render_partial_direct()` call.